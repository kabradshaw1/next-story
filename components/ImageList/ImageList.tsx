import Image from 'next/image';

type Props = {
  downloadURLs?: Array<string | null> | null;
  alt: string;
};

const ImageList: React.FC<Props> = ({ downloadURLs, alt }) => {
  return (
    <div className="gap-2">
      {downloadURLs?.map(
        (downloadURL) =>
          downloadURL !== null && (
            <Image
              key={downloadURL}
              src={downloadURL}
              alt={alt}
              width={200}
              height={200}
            />
          )
      )}
    </div>
  );
};

export default ImageList;
