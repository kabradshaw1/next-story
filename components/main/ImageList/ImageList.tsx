import Image from 'next/image';

type Props = {
  images: Array<{
    imageUrl: string;
    alt: string;
  }>;
};

const ImageList: React.FC<Props> = ({ images }) => {
  return (
    <div className="flex flex-wrap justify-center gap-2">
      {images.map((image) => (
        <Image
          key={image.imageUrl}
          src={image.imageUrl}
          alt={image.alt}
          width={200}
          height={200}
        />
      ))}
    </div>
  );
};

export default ImageList;
