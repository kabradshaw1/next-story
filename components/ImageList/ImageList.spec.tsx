import React from 'react';

import { render } from '@testing-library/react';

import ImageList from './ImageList';

describe('ImageList Component', () => {
  const mockDownloadURLs = [
    'http://example.com/image1.jpg',
    'http://example.com/image2.jpg',
  ];

  test('renders ImageList correctly', () => {
    const { container } = render(
      <ImageList downloadURLs={mockDownloadURLs} alt="Test Image" />
    );
    expect(container).toMatchSnapshot();
  });

  test('renders all images', () => {
    const { getAllByAltText } = render(
      <ImageList downloadURLs={mockDownloadURLs} alt="Test Image" />
    );

    const imgElements = getAllByAltText('Test Image');
    expect(imgElements).toHaveLength(mockDownloadURLs.length);
    imgElements.forEach((imgElement, index) => {
      expect(imgElement).toBeInTheDocument();
      expect(imgElement).toHaveAttribute(
        'src',
        expect.stringContaining(encodeURIComponent(mockDownloadURLs[index]))
      );
    });
  });

  test('renders without images', () => {
    const { container } = render(
      <ImageList downloadURLs={[]} alt="No Image" />
    );
    expect(container.firstChild).toBeEmptyDOMElement();
  });

  test('images have alt text', () => {
    const { getAllByAltText } = render(
      <ImageList downloadURLs={mockDownloadURLs} alt="Test Image" />
    );

    const imgElements = getAllByAltText('Test Image');
    expect(imgElements).toHaveLength(mockDownloadURLs.length);
    imgElements.forEach((imgElement) => {
      expect(imgElement).toBeInTheDocument();
    });
  });
});
