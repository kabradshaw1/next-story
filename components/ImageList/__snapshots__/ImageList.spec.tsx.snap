// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImageList Component renders ImageList correctly 1`] = `
<div>
  <div
    class="gap-2"
  >
    <img
      alt="Test Image"
      data-nimg="1"
      decoding="async"
      height="200"
      loading="lazy"
      src="/_next/image?url=http%3A%2F%2Fexample.com%2Fimage1.jpg&w=640&q=75"
      srcset="/_next/image?url=http%3A%2F%2Fexample.com%2Fimage1.jpg&w=256&q=75 1x, /_next/image?url=http%3A%2F%2Fexample.com%2Fimage1.jpg&w=640&q=75 2x"
      style="color: transparent;"
      width="200"
    />
    <img
      alt="Test Image"
      data-nimg="1"
      decoding="async"
      height="200"
      loading="lazy"
      src="/_next/image?url=http%3A%2F%2Fexample.com%2Fimage2.jpg&w=640&q=75"
      srcset="/_next/image?url=http%3A%2F%2Fexample.com%2Fimage2.jpg&w=256&q=75 1x, /_next/image?url=http%3A%2F%2Fexample.com%2Fimage2.jpg&w=640&q=75 2x"
      style="color: transparent;"
      width="200"
    />
  </div>
</div>
`;
